partitions feature documentation (#763)
canonical/craft-parts: docs/common/craft-parts/reference/partition_specific_output_directory_variables.rst

Partition-specific output directory environment variables
---------------------------------------------------------

> If partitions have been enabled, partition-specific environment variables will be created for use during step processing and execution of user-defined scriptlets.

Reviewer: ```suggestion
If partitions are enabled, partition-specific environment variables will be created for use during step processing and execution of user-defined scriptlets.
```
